### DISCUSSION TITLE
Why is it stopping after 25 deletions?

### DISCUSSION THREAD

**Discussion by githubUser01946** [victornpb/undiscord#590] [1. Questions & Answers]
Here are my settings currently:

![image](https://github.com/victornpb/undiscord/assets/117453319/ccb8a980-f815-4da2-9489-21b730adb67c)

In total there is 252 entries I am attempting to delete but it stops every 25 deletions so I have to keep running, is it a bug?

**Comment by Elivce** [comment]
Same thing happening to me

**Comment by doleperfection** [comment]
Same thing here, except it's not 25 deletions, it varies between 5-40. I've tried lots of different delays.

**Comment by TheActualLostar** [comment]
I am also having this issue; after 25 deletions, it stops. I have... over... 77k messages to delete. T_T

**Comment by LividJava** [comment]
You're hitting discord's rate limit. You're only allowed to delete (roughly) 25 messages/minute.

**Comment by jake983394** [comment]
does it say "ended because API returned an empty page" ?

**Reply by sententiaa** [reply]
It does, yes.

**Comment by rockythef0x** [comment]
There's a notice in the script that says it may be less reliable due to a Discord update. It's probably due to that. They're working on it (or at least, they say they are)

**Comment by i-hate-debugging** [comment]
One way around this is setting your delete delay so it deletes less than 25 messages a minute (or whatever the discord rate limit is). Currently I use 5450ms for delete delay and 2600ms for search delay. It takes longer but it is the only work around I've found.

**Reply by omoaubs** [reply]
this helps and works do this y'all!!!
